### ISSUE TITLE
Upgrading cluster after upgrading node version fails validation

### ISSUE THREAD

**Issue by hintofbasil** [weaveworks/eksctl#3727]
**What were you trying to accomplish?**

We were trying to upgrade out clusters from Kubernetes v1.18 to v1.19.

```bash
eksctl upgrade cluster \
    --config-file "/eks/${CLUSTER_NAME}.yaml" \
    --approve
```

We currently hard code the AMI version we use for each of our node groups.  This allows us fine grained control over when up grade our nodes.

While our cluster was running Kubernetes v1.18.9 the nodes were running v1.19.6.

**What happened?**

The cluster failed to update with the following error message (and log output).

```bash
2021-05-18 11:12:15 [ℹ]  eksctl version 0.45.0
2021-05-18 11:12:15 [ℹ]  using region eu-west-1
2021-05-18 11:12:15 [!]  NOTE: cluster VPC (subnets, routing & NAT Gateway) configuration changes are not yet implemented
2021-05-18 11:12:17 [!]  retryable error (Throttling: Rate exceeded
	status code: 400, request id: 3e23b4fb-1cf4-4bbe-aaed-9f05c8916a0f) from cloudformation/ListStacks - will retry after delay of 907.151748ms
2021-05-18 11:12:18 [!]  retryable error (Throttling: Rate exceeded
	status code: 400, request id: 22edc5e5-2dd9-476f-9f88-4805e17bfc5a) from cloudformation/ListStacks - will retry after delay of 571.589108ms
2021-05-18 11:12:18 [!]  retryable error (Throttling: Rate exceeded
	status code: 400, request id: 147d1f4e-3f6a-4f93-a0fe-d2ced2422970) from cloudformation/ListStacks - will retry after delay of 760.232877ms
2021-05-18 11:12:20 [ℹ]  will upgrade cluster "cluster-name" control plane from current version "1.18" to "1.19"
Error: InvalidParameterException: Nodegroups mixed-instances-8xl-45d1d1d3,nodes-default-gateway-ingress-5153e5bc must be updated to match cluster version 1.18 before updating cluster version
{
  RespMetadata: {
    StatusCode: 400,
    RequestID: "7b043638-ab1c-4fb8-ab25-4c37e8d5f5a8"
  },
  ClusterName: "cluster-name",
  Message_: "Nodegroups mixed-instances-8xl-45d1d1d3 must be updated to match cluster version 1.18 before updating cluster version"
}
```

It seems that `eksctl` checked the versions of the cluster and nodes, saw they were different and assumed the nodes were outdated however the nodes were actually a higher version.

I think `eksctl` should check the version and only error if the nodes are outdated.

**How to reproduce it?**

1. Create a Kubernetes 1.18 cluster and node groups using the following config (not validated after removing extra information).
2. Verify the nodes are using v1.19
3. Change the cluster version to '1.19'
4. Run the `eksctl upgrade cluster` command to try and upgrade the cluster

```yaml
cluster:
  apiVersion: eksctl.io/v1alpha5
  kind: ClusterConfig
  metadata:
    name: cluster-name
    region: eu-west-1
    version: '1.18'
  managedNodeGroups:
    - ami: ami-038341f2c72928ada
      availabilityZones:
        - eu-west-1a
      instanceTypes:
        - c5.9xlarge
        - c5a.8xlarge
        - c5d.9xlarge
        - c5n.9xlarge
        - m5.8xlarge
        - m5a.8xlarge
        - m5d.8xlarge
        - m5dn.8xlarge
        - m5n.8xlarge
        - r5.8xlarge
        - r5a.8xlarge
        - r5ad.8xlarge
        - r5d.8xlarge
        - r5dn.8xlarge
        - r5n.8xlarge
      maxSize: 50
      minSize: 1
      name: mixed-instances-8xl-45d1d1d3
      overrideBootstrapCommand: '#!/bin/bash

        yum install -y amazon-ssm-agent

        systemctl enable amazon-ssm-agent

        systemctl start amazon-ssm-agent

        rm -f /etc/systemd/system/kubelet.service.d/10-eksclt.al2.conf

        /etc/eks/bootstrap.sh cluster-name --kubelet-extra-args ''--cgroup-driver=cgroupfs
        --register-with-taints='''
      privateNetworking: true
      spot: true
      volumeSize: 128
```

**Anything else we need to know?**
eksctl running in Docker image based on `weaveworks/eksctl:0.45.0`

**Versions**
```
$ eksctl version
0.45.0

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-12T01:09:16Z", GoVersion:"go1.15.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.9-eks-d1db3c", GitCommit:"d1db3c46e55f95d6a7d3e5578689371318f95ff9", GitTreeState:"clean", BuildDate:"2020-10-20T22:18:07Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by hintofbasil** [comment]
We have realised we were using the wrong AMI id for the cluster version we were using.
We now have nodes running the same k8s version as our cluster.  Keeping this open though as it does feel like the logic here is incorrect - even if users shouldn't really find themselves in this situation.

**Comment by aclevername** [comment]
Thanks for opening the issue and the update @hintofbasil

Do you still believe the statement `It seems that eksctl checked the versions of the cluster and nodes, saw they were different and assumed the nodes were outdated however the nodes were actually a higher version.` is correct? That does sound like something worth fixing if so.

**Comment by hintofbasil** [comment]
Yes.  That's a good summary of the issue.

**Comment by nikimanoledaki** [comment]
> Thanks for opening the issue and the update @hintofbasil
> 
> Do you still believe the statement `It seems that eksctl checked the versions of the cluster and nodes, saw they were different and assumed the nodes were outdated however the nodes were actually a higher version.` is correct? That does sound like something worth fixing if so.

After doing some digging, I would reword the problem statement as: `It seems that EKS checked the versions of the cluster and nodes, saw they were different and assumed the nodes were outdated however the nodes were actually a higher version.`

The error is returned by EKS, specifically by [this](https://github.com/weaveworks/eksctl/blob/0dd8890c8e9bc46ed3b227ddb4cd671f103dbb29/pkg/eks/update.go#L285) call to EKS. It seems that EKS does some validation around the k8s version of the nodegroup versus that of the cluster and returns an error if the nodegroup k8s version is higher. After reproducing the steps in the description, I also tried to upgrade  the cluster through the console and got the same error.

**Comment by nikimanoledaki** [comment]
Closing this issue because this is intended behaviour - EKS does not recommend running a higher kube version on nodes compared to the cluster version. Also my previous comment about downgrading was wrong: downgrading nodegroups is possible by using a different custom AMI.
